hazelcast/src/main/java/com/hazelcast/map/impl/client/MapPortableHook.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.client;
+
+import com.hazelcast.internal.serialization.PortableHook;
+import com.hazelcast.internal.serialization.impl.FactoryIdHelper;
+import com.hazelcast.map.impl.querycache.accumulator.AccumulatorInfo;
+import com.hazelcast.nio.serialization.ClassDefinition;
+import com.hazelcast.nio.serialization.Portable;
+import com.hazelcast.nio.serialization.PortableFactory;
+
+import java.util.Collection;
+
+import static com.hazelcast.internal.serialization.impl.FactoryIdHelper.MAP_PORTABLE_FACTORY;
+import static com.hazelcast.internal.serialization.impl.FactoryIdHelper.MAP_PORTABLE_FACTORY_ID;
+
+/**
+ * Portable serialization hook for Map structures.
+ */
+public class MapPortableHook implements PortableHook {
+
+    public static final int F_ID = FactoryIdHelper.getFactoryId(MAP_PORTABLE_FACTORY, MAP_PORTABLE_FACTORY_ID);
+
+    public static final int CREATE_ACCUMULATOR_INFO = 1;
+
+    @Override
+    public int getFactoryId() {
+        return F_ID;
```

Comment:
While on EE side, this hook was using `FactoryIdHelper.ENTERPRISE_MAP_PORTABLE_FACTORY_ID` as its ID. I have switched it to the (unused until now) `FactoryIdHelper.MAP_PORTABLE_FACTORY_ID`. The `AccumulatorInfo` class is not used in client-member communication, so this change will not break compatibility with older clients.